gateway/gateway.go

```diff
@@ -106,6 +127,80 @@ func (g Gateway) HandleHTTPServiceRequest(ctx context.Context, httpReq *http.Req
 
 	// Use the gateway request context to process the relay(s) corresponding to the HTTP request.
 	// Any returned errors are ignored here and processed by the gateway context in the deferred calls.
-	// See the `BrodcastAllObservations` method of `gateway.requestContext` struct for details.
+	// See the `BroadcastAllObservations` method of `gateway.requestContext` struct for details.
 	_ = gatewayRequestCtx.HandleRelayRequest()
 }
+
+// handleWebsocketRequest handles WebSocket connection requests by directly connecting
+// to the provided websocket endpoint URL.
+//
+// Current Implementation:
+// - Bypasses protocol layer entirely as a temporary workaround
+// - Directly uses provided WebSocket endpoint URL
+// - Allows PATH to pass WebSocket messages without protocol support
+//
+// TODO_HACK(@commoddity, #143): Remove temporary workaround when Shannon protocol
+// supports WebSocket connections. Changes will:
+// - Utilize existing context system for endpoint selection
+// - Select from available Shannon protocol service endpoints
+// - Match HTTP request handling pattern
+// - Use HandleWebsocketRequest method defined on gateway.Protocol
+func (g Gateway) handleWebSocketRequest(ctx context.Context, httpReq *http.Request, w http.ResponseWriter) {
+	// Upgrade the HTTP request to a websocket connection.
+	// Do this first so that any errors that occur in the upgrade process can be sent
+	// to the websocket client as a close message, allowing easier debugging.
+	var upgrader = websocket.Upgrader{CheckOrigin: func(r *http.Request) bool { return true }}
+	clientConn, err := upgrader.Upgrade(w, httpReq, nil)
+	if err != nil {
+		g.Logger.Error().Msg("handleWebsocketRequest: error upgrading websocket connection request")
```

Comment:
Will this have metadata as to which one?

Reply:
Yes, once the `protocol` package method are implemented the logs will contain information about which endpoint is selected to handle the connection.

For now it does not since each service only has one websocket URL, which is set in the config YAML.